src/fs/file.rs

```diff
@@ -290,6 +291,56 @@ impl File {
         op.await
     }
 
+    /// Like [`read_at`], but using a pre-mapped buffer
+    /// registered with [`FixedBufRegistry`].
+    ///
+    /// [`read_at`]: Self::read_at
+    /// [`FixedBufRegistry`]: crate::buf::fixed::FixedBufRegistry
+    ///
+    /// # Errors
+    ///
+    /// In addition to errors that can be reported by `read_at`,
+    /// this operation fails if the buffer is not registered in the
+    /// current `tokio-uring` runtime.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    ///# fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// use tokio_uring::fs::File;
+    /// use tokio_uring::buf::fixed::FixedBufRegistry;
+    /// use tokio_uring::buf::IoBuf;
+    /// use std::iter;
+    ///
+    /// tokio_uring::start(async {
+    ///     let registry = FixedBufRegistry::new(iter::repeat(vec![0; 10]).take(10));
+    ///     registry.register()?;
+    ///
+    ///     let f = File::open("foo.txt").await?;
+    ///     let buffer = registry.check_out(2).unwrap();
+    ///
+    ///     // Read up to 10 bytes
+    ///     let (res, buffer) = f.read_fixed_at(buffer.slice(..), 0).await;
+    ///     let n = res?;
+    ///
+    ///     println!("The bytes: {:?}", &buffer[..n]);
+    ///
+    ///     // Close the file
+    ///     f.close().await?;
+    ///     Ok(())
+    /// })
+    ///# }
+    /// ```
+    pub async fn read_fixed_at(
+        &self,
+        buf: Slice<FixedBuf>,
+        pos: u64,
+    ) -> crate::BufResult<usize, Slice<FixedBuf>> {
```

Comment:
This signature has changed right? I thought it had been taking the id. This is probably better.

Reply:
Indeed: we have to pass valid buffer data to the op (the pointer, the length and the id), but the caller might want to limit the read to an offset portion of the buffer (e.g. to implement something like `read_exact`). With #163, the argument could be also generalized to `impl Into<Slice<FixedBuf>>`.

Reply:
Yes. If we end up having a choice, which would you like to see be merged first?